src/dfxvm/list.rs

```diff
@@ -1,20 +1,39 @@
+use crate::dfxvm::cli::ListOpts;
+use crate::dfxvm::manifest::Manifest;
 use crate::error::dfxvm::ListError;
+use crate::json::fetch_json;
 use crate::locations::Locations;
 use crate::settings::Settings;
 use itertools::Itertools;
+use reqwest::Url;
 use semver::Version;
 
-pub fn list(locations: &Locations) -> Result<(), ListError> {
+pub async fn list(opts: ListOpts, locations: &Locations) -> Result<(), ListError> {
     let settings = Settings::load_or_default(&locations.settings_path())?;
-    let default_version = settings.default_version;
+    if opts.available {
+        let url = Url::parse(&settings.manifest_url())?;
 
-    for version in installed_versions(locations)? {
-        let default_indicator = if default_version.as_ref() == Some(&version) {
-            " (default)"
-        } else {
-            ""
-        };
-        println!("{}{}", version, default_indicator);
+        info!("fetching {url}");
+        let manifest = fetch_json::<Manifest>(&url).await?;
+
+        if !manifest.versions.is_empty() {
+            let count = std::cmp::min(opts.limit, manifest.versions.len());
+            let versions = manifest.versions.iter().rev().take(count);
+            for version in versions {
+                println!("{}", version);
+            }
+        }
```

Comment:
This should work without the empty check
```suggestion
        let count = std::cmp::min(opts.limit, manifest.versions.len());
        let versions = manifest.versions.iter().rev().take(count);
        for version in versions {
            println!("{}", version);
        }
```

Reply:
Fixed at https://github.com/dfinity/dfxvm/pull/89/commits/6bf6a71f54c29ded886d079bb1ee2d95b3756d54